### DISCUSSION TITLE
Espressif ESP32-C6 and ESP32-H2 radio SoC support via ESP-Zigbee-SDK?

### DISCUSSION THREAD

**Discussion by Hedda** [ffenix113/zigbee_home#7] [Ideas]
Are you considering looking into adding support for ESP32-C6 and ESP32-H2 SoCs from Espressif? 

https://github.com/espressif/esp-zigbee-sdk

Espressif Zigbee SDK is also based on ZBOSS based (similar to Nordic Semiconductor Zigbee SDK), and believe they support Zephyr too?

https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/

FYI, ESPHome has an open feature request discussion here requesting Zigbee support in ESPHome on  ESP32-C6 and ESP32-H2 SoCs:

https://github.com/esphome/feature-requests/issues/1397

Both the ESP32-C6 SoC and the ESP32-H2 SoC models feature embedded IEEE 802.15.4 radio for native Zigbee (or Thread) support. With the ESP32-C6 being specifically optimized for mains-powered and the ESP32-H2 is optimized  for battery-powered operations. Espressif now has an "ESP Zigbee SDK" (based on same [ZBOSS](https://dsr-zoi.com/) Zigbee 3.0 stack as Nordic Semi's Zigbee implementation is based on), though their SDK is built on top of either [ESP-IDF](https://github.com/espressif/esp-idf) or [Arduino core for the ESP32](https://github.com/espressif/arduino-esp32) (both of which can also be utilized for ESPHome), that in turn depends on ESP ZBOSS 3.0 binary libraries, have a custom API, as well as custom CLI called EZNP (Espressif Zigbee Network Core-Processor Protocol), however, it should at least also be supported by the Zephyr project:

https://github.com/espressif/esp-zigbee-sdk

* https://components.espressif.com/components/espressif/esp-zigbee-lib

https://github.com/espressif/esp-zboss-lib

* https://components.espressif.com/components/espressif/esp-zboss-lib

https://github.com/espressif/esp-protocols

https://github.com/espressif/zephyr-toolchain

https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/

https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/developing.html#zigbee-product

https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/api-reference/esp_zigbee_core.html

https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/api-reference/zcl/index.html

https://docs.espressif.com/projects/esp-zigbee-sdk/en/latest/esp32/

https://components.espressif.com/components/espressif/esp-zigbee-lib

https://components.espressif.com/components/espressif/esp-zboss-lib

https://www.esp32.com/viewtopic.php?t=24624

https://dsr-zboss.com/#!/updates/dsr-joins-cmgc-partnership-s2link

https://www.prweb.com/releases/espressif_joins_zboss_open_initiative_zoi_a_community_for_royalty_free_zigbee_pro_stack_software/prweb18263163.htm

https://www.prweb.com/releases/announcing_launch_of_zboss_open_initiative_zoi_for_royalty_free_zigbee_pro_stack_software/prweb16814352.htm

**Comment by ffenix113** [comment]
This project is still in early days, and for now I want to focus on support for Nordic chips.
Still, if it would be possible to integrate support for ESP chips later on - it can be a worthy addition.

**Reply by tbrasser** [reply]
In that case, would it be possible to merge with esphome? There is no limit on esphome being esp based or wifi/ethernet based per-se. In a similar fashion support for tuya mcu's was added when libretiny merged into esphome.

**Reply by Hedda** [reply]
FYI, see this feature request which asks for native Zigbee device support inside ESPHome -> https://github.com/esphome/feature-requests/issues/1397

**Comment by Hedda** [comment]
FYI, looks like [P-R-O-C-H-Y](https://github.com/P-R-O-C-H-Y)'s Zigbee API (Zigbee library wrapper) for Arduino on ESP32 ([arduino-esp32](https://github.com/espressif/arduino-esp32)) is being working on very activly right now and [initial (basic) Zigbee library support has been merged](https://github.com/espressif/arduino-esp32/pull/10265) into both `master` and `release/v3.1` branches of upstream [arduino-esp32](https://github.com/espressif/arduino-esp32), so initial support and a few examples will be included in upcoming 3.0.6 and 3.1.0-rc2 releases of arduino-esp32. See:

* https://github.com/espressif/arduino-esp32/issues/10135

Note that even initial Zigbee API support is merged into upstream [arduino-esp32](https://github.com/espressif/arduino-esp32) is still in relativelyearly development so I would not expect it to be feature complete or cover all use cases needs as of yet, however this new Zigbee API is potentially the beginning of something that could become awesome of it grows and more developers join [P-R-O-C-H-Y](https://github.com/P-R-O-C-H-Y) in the development of it.

Initial tasks:

- https://github.com/espressif/esp32-arduino-lib-builder/pull/146
- https://github.com/espressif/arduino-esp32/pull/8844
- https://github.com/espressif/arduino-esp32/pull/9024 - https://github.com/espressif/arduino-esp32/pull/9627
